### DISCUSSION TITLE
Formatting on save, Scala 3 and SBT

### DISCUSSION THREAD

**Discussion by reidrac** [scalameta/metals#4142] [Q&A] [answered]
Hi,

I've started to use Scala 3 and other than some syntax highlighting issues with tree-sitter, all seems to work aces!

My only issue is that I have configured nvim to use metals to autoformat files on save, and when doing it for sbt files, it uses Scala 3 syntax that seems to break things.

I don't know if there's something obvious that I'm missing. I can NOT autoformat .sbt files, but sometimes there are sbt files with .scala extension, which makes things a bit complicated.

Any pointers/ideas would be appreciated!

Thank you.

Juan

**Comment by ckipp01** [comment]
> I don't know if there's something obvious that I'm missing. I can NOT autoformat .sbt files, but sometimes there are sbt files with .scala extension, which makes things a bit complicated.

Ahh, this is a good catch. I can reproduce this with just triggering a format in a Scala 3 project when there is a Scala file in the `project` dir and I have a setting like:

```
rewrite.scala3.removeOptionalBraces = true
```
Which indeed will format the `project/*.scala` file in a way that will cause the re-import to fail. This is actually something we'll need to fix in Metals itself. I'll create an issue with this discussion over there.

**Reply by ckipp01** [reply]
TIL you can transfer full discussions to other repos. That's cool.

**Reply by reidrac** [reply]
Thank you Chris!

**Comment by keynmol** [comment, marked as answer]
Going to cross-post here as well - @reidrac you can ask Scalafmt to apply a different dialect in different files, see config in my comment: https://github.com/scalameta/metals/issues/4143#issuecomment-1182479972

**Reply by reidrac** [reply]
Thank you @keynmol

**Reply by reidrac** [reply]
I used this in my `.scalafmt.conf` as per @keynmol recommendation and it solved the issue for me:
```
fileOverride {
  "glob:**.sbt" {
    runner.dialect = scala212source3
  }

  "glob:**/project/**.*" {
    runner.dialect = scala212source3
  }
}
```

**Reply by ckipp01** [reply]
@keynmol is the real MVP 🥇
